### PR TITLE
Support numeric hex colors for languages like JS

### PR DESCRIPTION
**Pull request by robinrodricks**
Hi there.

This is my first PR, I hope its okay.

I wanted to add support for numeric hex colors like `0xFFAABB` or `0xFFFFAABB` seen in languages like JavaScript, TypeScript, and others.

I've tested the Regex in a regex tool and it works.

**Changes:**

1. Add support for numeric hex colors using `numhex` and `numhex+alpha` tags
3. Add smart defaults in the extension settings for various common web programming files (HTML, JS, TS, CS, Java)
4. Update the readme based on these ([preview here](https://github.com/robinrodricks/vscode-color-info#colorinfolanguages))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,7 +110,39 @@
...
-            }
+            },
+            {
+              "selector": "html",
+              "colors": "css"
+            },
+		        {
+		        	"selector": "js",
+		        	"colors": [
+		        		"numhex",
```
Comment (mattbierner) [inline package.json:121]: Let's keep `numhex` disabled by default. They are likely to match on too many existing hex literals which are not for colors

File: README.md [hunk 2]
```diff
@@ -42,25 +42,7 @@ Array of fields not to display. Overrides `colorInfo.fields`. If you only need t
 
 
 #### `colorInfo.languages`
-Defines which files and languages have color info enabled, and which color types are supported. The default language setting enables all css color values for `css`, `sass`, and `less` documents:
-
```
Comment (mattbierner) [inline README.md:46]: This section is useful as it show users how they can add support for additional languages

File: README.md [hunk 3]
```diff
@@ -70,17 +52,19 @@ Each element consists of:
...
     * `hex+alpha` - Css hex plus alpha (`#ff00ff77` or `#f0f7`)
-    * `rgb` - Css rgb or rgba (`rgb(1, 2, 3)` or `rgba(1, 2, 3, 0.5)`) c
+    * `numhex` - Numeric hex (`0xFFAABB`)
+    * `numhex+alpha` - Numeric hex plus alpha (`0xFFAABBEE`)
+    * `rgb` - Css rgb or rgba (`rgb(1, 2, 3)` or `rgba(1, 2, 3, 0.5)`) 
     * `hsl` - Css hsl or hsla (`hsl(1, 2, 3)` or  `hsla(1, 2, 3, 0.5)`)
     * `css-colors-names` - Css color names (`red`, `blue`)
 
-VSCode does not currently support nested languages, so to enable Color Info in an `html` file, you must add:
+The default language setting enables all color values for the following file types:
```
Comment (mattbierner) [inline README.md:61]: Also feeling this should be reverted (at at least submitted as a separate PR)

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "color-info",
   "displayName": "Color Info",
-  "description": "Provides quick information about css colors",
+  "description": "Provides quick information about colors in any programming language",
```
Comment (mattbierner) [inline package.json:4]: Can you also please revert this change this it is not strictly related to the PR

File: package.json [hunk 5]
```diff
@@ -110,7 +110,39 @@
...
+		        	]
+		        },
+		        {
+		        	"selector": "cs",
+		        	"colors": [
+		        		"numhex",
+		        		"numhex+alpha"
+		        	]
+		        },
+		        {
```
Comment (mattbierner) [inline package.json:139]: Lower in the file, make sure to add the new number formats under `selector` too

File: README.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # VSCode Color Info
 
-[Visual Studio Code](https://code.visualstudio.com) extension that provides quick information css colors.
+[Visual Studio Code](https://code.visualstudio.com) extension that provides quick information of colors in any programming language.
```
Comment (mattbierner) [inline README.md:3]: Can you also please revert this change this it is not strictly related to the PR

File: src/color_extractor.ts [hunk 7]
```diff
@@ -83,6 +83,26 @@ const hexaExtractor: ColorValueExtractor = {
     },
 };
 
+/**
+ * Get all numeric hex colors in a line of text.
+ */
+const numHexExtractor: ColorValueExtractor = {
```
Comment (mattbierner) [inline src/color_extractor.ts:89]: Do you also have to update the `ColorValueExtractorType` type?

**Review by mattbierner — CHANGES_REQUESTED** [review]
Code changes look good but let's revert some of the unrelated readme changes

Let me know if you have any questions about the comments

**Comment by robinrodricks** [comment]
Nevermind.
